packages/query/src/block-processor.ts

```diff
@@ -419,29 +419,42 @@ export class BlockProcessor implements BlockProcessorInterface {
   private async identifyAuctionNfts(action: Action['action']) {
     if (action.case === 'actionDutchAuctionSchedule' && action.value.description) {
       const auctionId = getAuctionId(action.value.description);
-      const metadata = getAuctionNftMetadata(
-        auctionId,
-        // Always a sequence number of 0 when starting a Dutch auction
-        0n,
-      );
+
+      // Always a sequence number of 0 when starting a Dutch auction
+      const seqNum = 0n;
+
+      const metadata = getAuctionNftMetadata(auctionId, seqNum);
+
       await Promise.all([
         this.indexedDb.saveAssetsMetadata(metadata),
         this.indexedDb.upsertAuction(auctionId, {
           auction: action.value.description,
+          seqNum,
         }),
       ]);
     } else if (action.case === 'actionDutchAuctionEnd' && action.value.auctionId) {
-      const metadata = getAuctionNftMetadata(
-        action.value.auctionId,
-        // Always a sequence number of 1 when ending a Dutch auction
-        1n,
-      );
-      await this.indexedDb.saveAssetsMetadata(metadata);
+      // Always a sequence number of 1 when ending a Dutch auction
+      const seqNum = 1n;
+
+      const metadata = getAuctionNftMetadata(action.value.auctionId, seqNum);
```

Comment:
won't this request metadata for closed auctions since it's passing in `seqNum` of 1? Shouldn't it pass `seqNum` of 0 for active auctions, and only after save the metadata and update the auction sequence number? Or does it not matter here?

Reply:
@TalDerei 

> won't this request metadata for closed auctions since it's passing in seqNum of 1?

Correct — although it's not exactly _requesting_ the metadata, but rather _generating_ it. (`getAuctioNftMetadata()` calls a WASM function under the hood that generates metadata for an asset that doesn't exist yet.)

> Shouldn't it pass seqNum of 0 for active auctions, and only after save the metadata and update the auction sequence number?

Hm — we could request the NFT metadata for the open auction, and the modify its sequence number. But that feels a bit messy to me — if the algorithm for generating NFT metadata changes in the future such that closed-auction NFTs have different metadata than open-auction NFTs, this would break. Easier to just generate it from scratch for each auction state.

Reply:
yeah was thinking about that, I think generating the metadata is the right design choice here.

Reply:
> requesting

yeah wrong word choice here 😅